### PR TITLE
Fix the trace unstable validation issue

### PR DESCRIPTION
**Pull request by wyTrivail**

### CODE REVIEW

File: validator/src/main/resources/expected-data-template/spark/sparkAppExpectedAWSSDKTrace.mustache [hunk 1]
```diff
@@ -16,35 +16,7 @@
     {
       "fault": false,
       "error": false,
-      "throttle": false,
-      "http": {
-        "request": {
```
Comment (bhautikpip) [inline validator/src/main/resources/expected-data-template/spark/sparkAppExpectedAWSSDKTrace.mustache:21]: Why are we removing all of it?

File: validator/src/main/java/com/amazon/aoc/validators/TraceValidator.java [hunk 2]
```diff
@@ -88,61 +88,56 @@ public void validate() throws Exception {
...
         () -> {
           List<Trace> retrieveTraceList = null;
           retrieveTraceList = xrayService.listTraceByIds(traceIdList);
-          retrieveTraceListAtomicReference.set(retrieveTraceList);
-
           if (retrieveTraceList == null || retrieveTraceList.isEmpty()) {
             throw new BaseException(ExceptionCode.EMPTY_LIST);
           }
+
+          // in case the json format is wrong, retry it.
```
Comment (bhautikpip) [inline validator/src/main/java/com/amazon/aoc/validators/TraceValidator.java:102]: What do you mean by json format is wrong?

Comment (wyTrivail) [reply validator/src/main/java/com/amazon/aoc/validators/TraceValidator.java:102]: check this: https://github.com/aws-observability/aws-otel-collector/runs/1397013128?check_suite_focus=true, the "e2etest-ecs (otlp_trace, EC2)" one, looks like xray returned a empty/invalid json of segment. but i can't reproduce it on my local even with the same trace id, so i suspect xray backend somehow returned a invalid json

File: validator/src/main/resources/expected-data-template/spark/sparkAppExpectedAWSSDKTrace.mustache (hunk 1 above)
Comment (wyTrivail) [reply validator/src/main/resources/expected-data-template/spark/sparkAppExpectedAWSSDKTrace.mustache:21]: the final one is 

```
[{
  "name": "aws-otel-integ-test",
  "fault": false,
  "error": false,
  "throttle": false,
  "http": {
    "request": {
      "url": "{{endpoint}}/aws-sdk-call",
      "method": "GET"
    },
    "response": {
      "status": 200
    }
  },
  "subsegments": [
    {
      "fault": false,
      "error": false,
      "throttle": false
    }
  ]
}]
```

the http block under subsegments are unpredictable, some times a put request is appeared

**Review by bhautikpip — APPROVED** [review]

**Pull request merged**